ethereum/erc20/insert_weth_balance_changes.sql

```diff
@@ -0,0 +1,279 @@
+SELECT dex.insert_weth_balance_changes('2021-07-05','2021-07-12') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-07-05' AND hour < '2021-07-12');
+SELECT dex.insert_weth_balance_changes('2021-07-12','2021-07-19') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-07-12' AND hour < '2021-07-19');
+SELECT dex.insert_weth_balance_changes('2021-07-19','2021-07-26') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-07-19' AND hour < '2021-07-26');
+SELECT dex.insert_weth_balance_changes('2021-07-26','2021-08-02') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-07-26' AND hour < '2021-08-02');
+SELECT dex.insert_weth_balance_changes('2021-08-02','2021-08-09') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-08-02' AND hour < '2021-08-09');
+SELECT dex.insert_weth_balance_changes('2021-08-09','2021-08-16') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-08-09' AND hour < '2021-08-16');
+SELECT dex.insert_weth_balance_changes('2021-08-16','2021-08-23') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-08-16' AND hour < '2021-08-23');
+SELECT dex.insert_weth_balance_changes('2021-08-23','2021-08-30') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-08-23' AND hour < '2021-08-30');
+SELECT dex.insert_weth_balance_changes('2021-08-30','2021-09-06') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-08-30' AND hour < '2021-09-06');
+SELECT dex.insert_weth_balance_changes('2021-09-06','2021-09-13') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-09-06' AND hour < '2021-09-13');
+SELECT dex.insert_weth_balance_changes('2021-09-13','2021-09-20') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-09-13' AND hour < '2021-09-20');
+SELECT dex.insert_weth_balance_changes('2021-09-20','2021-09-27') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-09-20' AND hour < '2021-09-27');
+SELECT dex.insert_weth_balance_changes('2021-09-27','2021-10-04') WHERE NOT EXISTS (SELECT * FROM dex.hourly_balance_changes WHERE hour >= '2021-09-27' AND hour < '2021-10-04');
+
+-- final fill
+SELECT dex.insert_weth_balance_changes(
+    '2021-10-04',
+    date_trunc('hour', now())
+)
+WHERE NOT EXISTS (
+    SELECT *
+    FROM dex.hourly_balance_changes
+    WHERE hour >= '2021-10-04'
+    AND hour < date_trunc('hour', now())
+);
+
+
+INSERT INTO cron.job (schedule, command)
+VALUES ('57 * * * *', $$
+    SELECT dex.insert_weth_balance_changes(
+        (SELECT (SELECT max(hour) FROM dex.hourly_balance_changes) + interval '1 hour'),
+        (SELECT date_trunc('hour', now())));
```

Comment:
Usually we do `now() - 20 minutes` because after 20 min we should be in sync. In this case maybe we should do up until the last hour or so, since we insert for every hour.

Reply:
Up until the last hour is what I tried to achieve by running the cron job at minute 57 and `truncating` back until the begin of the hour with  `date_trunc('hour', now())`. Or did I make a mistake in the logic?